src/api/hooks/useNavigateWithParams.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  createSearchParams,
+  generatePath,
+  useNavigate,
+  useSearchParams,
+} from "react-router-dom";
+import {defaultNetworkName} from "../../constants";
+
+export function useNavigateWithParams() {
```

Comment:
cc @jillxuu let's use `useNavigateWithParams` for future page navigation. This will ensure persistent network